pkg/rc/replication_controller_test.go

```diff
@@ -237,3 +237,39 @@ func TestUnschedule(t *testing.T) {
 	Assert(t).AreEqual(len(scheduled), 0, "expected a pod to have been unlabeled")
 	Assert(t).AreEqual(len(kp.manifests), 0, "expected manifest to have been unscheduled")
 }
+
+func TestPreferUnscheduleIneligible(t *testing.T) {
+	rcStore, kp, applicator, rc := setup(t)
+	err := applicator.SetLabel(labels.NODE, "node1", "nodeQuality", "good")
+	Assert(t).IsNil(err, "expected no error labeling node1")
+	err = applicator.SetLabel(labels.NODE, "node2", "nodeQuality", "good")
+	Assert(t).IsNil(err, "expected no error labeling node2")
+	err = applicator.SetLabel(labels.NODE, "node3", "nodeQuality", "good")
+	Assert(t).IsNil(err, "expected no error labeling node3")
+
+	quit := make(chan struct{})
+	rc.WatchDesires(quit)
+
+	rcStore.SetDesiredReplicas(rc.ID(), 3)
+	numNodes := waitForNodes(t, rc, 3)
+	Assert(t).AreEqual(numNodes, 3, "took too long to schedule")
+
+	scheduled := scheduledPods(t, applicator)
+	Assert(t).AreEqual(len(scheduled), 3, "expected a pod to have been labeled")
+	Assert(t).AreEqual(len(kp.manifests), 3, "expected a manifest to have been scheduled")
+
+	// Make node2 ineligible, so that it will be preferred for unscheduling
```

Comment:
I'm wondering a bit - if `node3` would have been the preferred node in either case, then this test will be inconclusive. So basically, depending on the order of `PopAny`, this test might falsely declare bad code good.

But I don't want to suggest that we have one test with `node2` ineligible and one test with `node3` ineligible. So I'm not sure we can solve that problem. So maybe we just have to live with it? Tests can't prove the absence of bugs, only their presence, or some expression to that effect?